src/Counter/Connection.php

```diff
@@ -0,0 +1,113 @@
+<?php
+
+namespace atk4\dsql\Counter;
+
+use atk4\dsql\Expression;
+use atk4\dsql\ProxyConnection;
+use atk4\dsql\Query;
+
+class Connection extends ProxyConnection
+{
+    const DEFAULT_DRIVER_TYPE = 'counter';
```

Comment:
Why not simply DRIVER_TYPE ?

Reply:
The actual driver type is in `$driverType` property.
`DEFAULT_DRIVER_TYPE` is the default one that will be used if no other driverType is set
I spend a few hours figuring out how to name it. Naming it `DRIVER_TYPE` does not hint to what the actual use is and what is the difference with the property `$driverType`

Reply:
What is the problem with setting `$driverType` to the default value?

Reply:
My idea was to have a way of retrieving the default driver type for a connection class without having to instantiate it. (e.g `MySQL\Connection::DEFAULT_DRIVER_TYPE`)
This is used in `Connection::register` where the connection class is registered for its default driver type if no other set specifically.

Reply:
I am not sure if that can be done using reflection, but probably yes :D

Reply:
https://www.php.net/manual/en/reflectionclass.getdefaultproperties.php

Reply:
Used your suggestion and got rid of the constant altogether

Reply:
👍